registry/client_factory.go

```diff
@@ -52,22 +76,17 @@ func (f *RemoteClientFactory) ClientFor(repo image.CanonicalName, creds Credenti
 	manager := f.challengeManager
 	f.mu.Unlock()
 
-	scheme := "https"
-	for _, h := range f.InsecureHosts {
-		if repo.Domain == h {
-			scheme = "http"
-		}
-	}
-
 	registryURL := url.URL{
-		Scheme: scheme,
+		Scheme: "https",
 		Host:   repo.Domain,
 		Path:   "/v2/",
 	}
 
 	// Before we know how to authorise, need to establish which
 	// authorisation challenges the host will send. See if we've been
 	// here before.
+	attemptInsecureFallback := insecure
+attempt:
```

Comment:
Maybe make the label more descriptive? (e.g. `schemeAttempt` or `attemptScheme`)